core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java

```diff
@@ -215,14 +227,112 @@ public boolean isAvailable() {
       return file.exists();
    }
 
+   private boolean isCorruptData() throws IOException {
+      ByteBuffer buf = ByteBuffer.allocate(KEY_POS_LATEST);
+      long pos = filePos;
+      buf = readChannel(buf, pos, KEY_POS_LATEST);
+      if (buf.remaining() > 0)
+         return false;
+      buf.flip();
+      FileEntry fe = new FileEntry(pos, buf);
+      // Explicitly calculate the correct size as incorrect value stored in fe.size
+      int actualSize = KEY_POS_LATEST + fe.keyLen + fe.dataLen + fe.metadataLen + fe.internalMetadataLen;
```

Comment:
isn't the `expiryTime` missing? 
can't we double check that `actualSize == fe.size`?

Reply:
ah, `expiryTime` is in `KEY_POS_LATEST` already :)

Reply:
> can't we double check that `actualSize == fe.size`?

No as `fe.size` was incorrectly set in the old migration code. Instead of using `.size` of the stored entry, it actually re-used the old entry that it was migrating.